### ISSUE TITLE
Rule duplication `F509` and `PLE1300`

### ISSUE THREAD

**Issue by dhruvmanila** [astral-sh/ruff#11403]
Both rules check for invalid format character in `%` format style string.

The implementation is almost the same:

### `F509`
https://github.com/astral-sh/ruff/blob/1f748b89ca73361146508b6f0af9afc756dc9cb2/crates/ruff_linter/src/checkers/ast/analyze/expression.rs#L1087-L1100

### `PLE1300`
https://github.com/astral-sh/ruff/blob/1f748b89ca73361146508b6f0af9afc756dc9cb2/crates/ruff_linter/src/rules/pylint/rules/bad_string_format_character.rs#L117-L125

There's one difference which can be noticed when an implicitly concatenated string is used. `F509` looks at the concatenated string while `PLE1300` looks at each part of an implicitly concatenated string. Refer to https://play.ruff.rs/6efd9945-2a3b-43e4-b9b7-a8d041b98318. 

Reference:
* `F509` https://docs.astral.sh/ruff/rules/percent-format-unsupported-format-character/
* `PLE1300` https://docs.astral.sh/ruff/rules/bad-string-format-character/

**Comment by charliermarsh** [comment]
@dhruvmanila - Let's remove `PLE1300` as part of v0.5.0?

**Comment by zanieb** [comment]
See https://github.com/astral-sh/ruff/pull/9756 for prior art.

**Comment by dhruvmanila** [comment]
> Let's remove `PLE1300` as part of v0.5.0?

Yeah, that makes sense but there's one difference where `PLE1300` would check each part of an implicitly concatenated string while `F509` checks it as a whole. This means that the number of diagnostics would be reduced. That said, I think `F509` can be improved to either 
1. Include all the invalid characters in the message or
2. Raise violation for each invalid character and highlighting the part of the string instead of the entire string

**Comment by dhruvmanila** [comment]
I'm looking into this and it seems that there's one behavior which is different - `PLE1300` looks at _both_ `%` style strings and `.format` strings while `F509` only looks at `%` style strings. Basically, `F509` is a subset of `PLE1300` which means we should remove `F509` instead.

**Comment by Pierre-Sassoulas** [comment]
Shouldn't they be aliased to match what's expected from each tool and not break rétro compatibility ? (There's another issue for human readable names where old_names from pylint and aliases came up).

**Comment by dhruvmanila** [comment]
We've decided to not do this via rule redirection but wait until rule re-categorization (#1774) is complete and merge these rules instead.
